lib/fluent/config/element.rb

```diff
@@ -144,6 +142,42 @@ def secret_param?(key)
         false
       end
 
+      def param_type(key)
+        return nil if @corresponding_proxies.empty?
+
+        param_key = key.to_sym
+        proxy = @corresponding_proxies.detect do |_proxy|
+          _proxy.params.has_key?(param_key)
+        end
+        return nil unless proxy
+        _block, opts = proxy.params[param_key]
+        opts[:type]
+      end
+
+      def dump_value(k, v, indent, nindent)
```

Comment:
Is there any reason to assign `out = ""` at once and then do `out << "..."`?
It looks good enough just to return string value by `if` expression.